In [1]:
import pandas as pd

df = pd.read_csv("hf://datasets/zefang-liu/phishing-email-dataset/Phishing_Email.csv")

In [2]:
df.head(20)

Unnamed: 0                                         Email Text  \
0            0  re : 6 . 1100 , disc : uniformitarianism , re ...   
1            1  the other side of * galicismos * * galicismo *...   
2            2  re : equistar deal tickets are you still avail...   
3            3  \nHello I am your hot lil horny toy.\n    I am...   
4            4  software at incredibly low prices ( 86 % lower...   
5            5  global risk management operations sally congra...   
6            6  On Sun, Aug 11, 2002 at 11:17:47AM +0100, wint...   
7            7  entourage , stockmogul newsletter ralph velez ...   
8            8  we owe you lots of money dear applicant , afte...   
9            9  re : coastal deal - with exxon participation u...   
10          10  make her beg you to give it to her everynight ...   
11          11  URL: http://www.newsisfree.com/click/-5,830431...   
12          12  begin forwarded text Date: Wed, 25 Sep 2002 13...   
13          13  re : fyi - wellhead portfolio who is considere...   
14          14  rmmla / ads * * * * * * * * papers solicited f...   
15          15  re : testing ir & fx var nick and winston , i ...   
16          16  The academic discipline of Software Engineerin...   
17          17  re : 3 . 402 queries : language - speakers , s...   
18          18  a resume john , this is a resume i received to...   
19          19  EFFector       Vol. 15, No. 35       November ...   

        Email Type  
0       Safe Email  
1       Safe Email  
2       Safe Email  
3   Phishing Email  
4   Phishing Email  
5       Safe Email  
6       Safe Email  
7   Phishing Email  
8   Phishing Email  
9       Safe Email  
10  Phishing Email  
11      Safe Email  
12      Safe Email  
13      Safe Email  
14      Safe Email  
15      Safe Email  
16      Safe Email  
17      Safe Email  
18      Safe Email  
19      Safe Email

In [3]:
len(df)

18650

In [4]:
df.shape

(18650, 3)

In [5]:
df.columns

Index(['Unnamed: 0', 'Email Text', 'Email Type'], dtype='object')

3 columns, however, first column is just row number and is labeled "unnamed" should remove.

In [6]:
df.dtypes

Unnamed: 0     int64
Email Text    object
Email Type    object
dtype: object

In [7]:
df.isnull().sum()  

Unnamed: 0     0
Email Text    16
Email Type     0
dtype: int64

16 rows that have missing Email Text values. Need to remove. 

In [8]:
df.duplicated().sum()

np.int64(0)

No duplicates rows found

In [9]:
df.nunique()

Unnamed: 0    18650
Email Text    17537
Email Type        2
dtype: int64

There are 17,537 unique emails out of 18,650 rows
1,113 duplicates

In [10]:
df.duplicated(subset=['Email Text']).sum()

np.int64(1112)

(counts all rows that are duplicates after the first occurrence)
remove 1,112 rows

In [17]:
df.columns

Index(['Unnamed: 0', 'Email Text', 'Email Type'], dtype='object')

In [ ]:
import re
from collections import Counter

# Optional: clean text inline (recommended)
def clean_text(s):
    s = str(s).lower()
    s = re.sub(r'http\S+', ' ', s)      # remove URLs
    s = re.sub(r'[^a-z\s]', ' ', s)     # remove punctuation/numbers
    s = re.sub(r'\s+', ' ', s).strip()  # normalize spaces
    return s

# Clean if not already cleaned
df['Email Text'] = df['Email Text'].apply(clean_text)

# Separate phishing vs safe emails
phish_texts = " ".join(df.loc[df['Email Type'] == 'Phishing Email', 'Email Text']).split()
safe_texts  = " ".join(df.loc[df['Email Type'] == 'Safe Email', 'Email Text']).split()



In [24]:
phish_df = df[df['Email Type'] == 'Phishing Email']


In [25]:
phish_df.head()

Unnamed: 0                                         Email Text  \
3            3  hello i am your hot lil horny toy i am the one...   
4            4  software at incredibly low prices lower draper...   
7            7  entourage stockmogul newsletter ralph velez ge...   
8            8  we owe you lots of money dear applicant after ...   
10          10  make her beg you to give it to her everynight ...   

        Email Type  
3   Phishing Email  
4   Phishing Email  
7   Phishing Email  
8   Phishing Email  
10  Phishing Email

In [22]:

# Combine all phishing emails into one long string of words
phish_words = " ".join(phish_df['Email Text']).split()

# Count how often each word appears
phish_counts = Counter(phish_words)

# Get the 30 most common phishing keywords
top_phish = phish_counts.most_common(30)
print(top_phish)


[('the', 57450), ('to', 46511), ('and', 39024), ('of', 34924), ('you', 31915), ('a', 29841), ('in', 22429), ('your', 19743), ('for', 19721), ('this', 17886), ('is', 17587), ('i', 12354), ('that', 11419), ('with', 10679), ('on', 10459), ('it', 10428), ('or', 9984), ('be', 9936), ('are', 9507), ('we', 9485), ('s', 9120), ('from', 8513), ('will', 8339), ('have', 8290), ('as', 8243), ('our', 8031), ('not', 7487), ('all', 6632), ('by', 6224), ('e', 5920)]
